java-checks/src/main/java/org/sonar/java/checks/unused/UnusedMethodParameterCheck.java

```diff
@@ -80,7 +89,34 @@ private static boolean isExcluded(MethodTree tree) {
   }
 
   private static boolean isAnnotated(MethodTree tree) {
-    return !tree.modifiers().annotations().isEmpty();
+    if (tree.modifiers().annotations().isEmpty()) {
+      return false;
+    }
+
+    // If any annotation doesn't match the @SuppressWarning then mark the method as annotated.
+    return tree.modifiers().annotations().stream().anyMatch(annotation -> !isExcludedSuppressWarningAnnotation(annotation));
+  }
+
+  private static boolean isExcludedSuppressWarningAnnotation(AnnotationTree tree) {
+    if (!tree.annotationType().symbolType().is(SUPPRESS_WARNINGS_ANNOTATION)) {
+      return false;
+    }
+
+    return tree.arguments().stream().allMatch(UnusedMethodParameterCheck::containsOnlyExcludedWarnings);
+  }
+
+  private static boolean containsOnlyExcludedWarnings(Tree tree) {
+    if (tree.is(Tree.Kind.STRING_LITERAL)) {
+      return EXCLUDED_WARNINGS_SUPPRESIONS.contains(((LiteralTree) tree).value());
+    } else if (tree.is(Tree.Kind.NEW_ARRAY)) {
+      return ((JavaTree) tree).getChildren().stream()
```

Comment:
There are two problems here : usage of stream API is a bit overkill and JavaTree is not part of the API but rather an implementation detail, so I would suggest to cast tree to NewArrayTree and have a for loop over its initializers (or use stream API here). That will end up in the same result but more explicit and without the use of downcast to JavaTree.